examples/all-clusters-app/linux/AllClustersCommandDelegate.h

```diff
@@ -30,7 +30,7 @@ class AllClustersAppCommandHandler
 
     static void HandleCommand(intptr_t context);
 
-    AllClustersAppCommandHandler(Json::Value jasonValue) : mJsonValue(std::move(jasonValue)) {}
+    AllClustersAppCommandHandler(Json::Value & jasonValue) : mJsonValue(std::move(jasonValue)) {}
```

Comment:
will this force the code call move twice?  since we need to use move explicitly when we create AllClustersAppCommandHandler instance.

```
    Json::Value value;
    return Platform::New<AllClustersAppCommandHandler>(std::move(value));
    
AllClustersAppCommandHandler(Json::Value && jasonValue) : mJsonValue(std::move(jasonValue)) {}
```

move use shallow copy which has less cost, but passing by reference has no cost.

Reply:
move is a pass by reference until the actual move constructor runs.  `std::move` is just a cast; it doesn't do any copying.

So yes, this should be taking an rvalue reference as @mrjerryjohns suggests.

Reply:
But we need to call std::move twice in this case, one for cast, and another one for moving

Reply:
std::move is _always a cast_.  That's all it is.